merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/generator/MapperMethodMaker.java

```diff
@@ -4,48 +4,48 @@
 import com.squareup.javapoet.MethodSpec;
 import com.squareup.javapoet.ParameterizedTypeName;
 import com.squareup.javapoet.TypeName;
-import gov.nasa.jpl.aerie.merlin.processor.metamodel.ExportDefaultsStyle;
 import gov.nasa.jpl.aerie.merlin.processor.metamodel.ActivityTypeRecord;
 import gov.nasa.jpl.aerie.merlin.processor.metamodel.ExportTypeRecord;
 import gov.nasa.jpl.aerie.merlin.protocol.model.ConfigurationType;
 import gov.nasa.jpl.aerie.merlin.protocol.model.TaskSpecType;
 import gov.nasa.jpl.aerie.merlin.protocol.types.MissingArgumentException;
-import gov.nasa.jpl.aerie.merlin.processor.metamodel.ParameterRecord;
 import gov.nasa.jpl.aerie.merlin.protocol.types.Parameter;
 
-import javax.lang.model.element.ElementKind;
 import javax.lang.model.element.Modifier;
-import javax.lang.model.element.TypeElement;
-import java.util.ArrayList;
 import java.util.List;
 import java.util.stream.Collectors;
 
-public interface MapperMethodMaker {
+public abstract sealed class MapperMethodMaker permits
+    AllDefinedMethodMaker,
+    AllStaticallyDefinedMethodMaker,
+    NoneDefinedMethodMaker,
+    SomeStaticallyDefinedMethodMaker
+{
+  protected final ExportTypeRecord exportType;
```

Comment:
Just so I can understand - protected means the field is accessible by any class within this package, as well as by subclasses of this class in other packages?

If all of the sealed subclasses are in this package, would it make sense for these to be package private? (Happy to hear a counter-argument - I haven't used `protected` much before)

Reply:
Ahh true, I always forget that `protected` is less strict than package private. I'll make this package private :)

Reply:
Just pushed change for this